pkg/datastore/mysql/query.go

```diff
@@ -74,25 +81,72 @@ func buildWhereClause(filters []datastore.ListFilter) string {
 	return fmt.Sprintf("WHERE %s", strings.Join(conds[:], " AND "))
 }
 
-func buildOrderByClause(orders []datastore.Order) string {
-	if len(orders) == 0 {
+func buildPaginationCondition(opts datastore.ListOptions) string {
+	// Skip on no cursor.
+	if len(opts.Cursor) == 0 {
 		return ""
 	}
 
+	conds := make([]string, len(opts.Orders))
+	for i, o := range opts.Orders {
+		if o.Field == "Id" {
+			conds[i] = fmt.Sprintf("%s %s UUID_TO_BIN(?,true)", o.Field, makePaginationConditionOperator(o))
+		} else {
+			conds[i] = fmt.Sprintf("%s %s ?", o.Field, makePaginationConditionOperator(o))
+		}
+	}
+
+	// If there is no filter, mean pagination condition should be treated as the only where condition.
+	if len(opts.Filters) == 0 {
+		return fmt.Sprintf("WHERE %s", strings.Join(conds[:], " AND "))
+	}
+	return fmt.Sprintf("AND %s", strings.Join(conds[:], " AND "))
+}
+
+func makePaginationConditionOperator(order datastore.Order) string {
+	var ope string
+	// Only the Id field should be used strict with greater/lower than operators.
+	if order.Field == "Id" {
```

Comment:
Got the magic here. Not including the "=" from the last order. 😄

Reply:
You named it 😁